lib/Sema/CSSimplify.cpp

```diff
@@ -4419,6 +4419,14 @@ ConstraintSystem::matchTypesBindTypeVar(
                : getTypeMatchFailure(locator);
   }
 
+  // If typeVar is key path value, leave unsolved until it is directly bound.
```

Comment:
- I think this needs to be wrapped in `if (shouldAttemptFixes())`
- For the comment, how about - `if type variable represents a key path value type, differ binding it to contextual type in diagnostic mode. We want it to be bound from the last key path component to help with diagnostics.`